.github/workflows/nightly-noinstall.yml

```diff
@@ -2,6 +2,10 @@ name: OSM NoInstall Nightly Job
 on: 
   schedule:
     - cron: "0 0 * * *"
+  pull_request:
+    branches:
+      - main
+      - release-*
```

Comment:
We do not want to be running this for every pull request till this is stable, at which point it can be augmented into the default CI workflow for PRs.

Reply:
@shashankram I just added this temporarily for this PR so I can run it every time I push a change. Once it starts running without any errors I'll remove the "pull request" part and just keep it as a nightly job.

Reply:
Sounds good. For such PRs, so people don't start reviewing, make the PR a draft/wip and put the `do-not-merge/hold` label.